### DISCUSSION TITLE
Collision detection with 2 robots

### DISCUSSION THREAD

**Discussion by mbusy** [NVlabs/curobo#153] [Software Q&A] [answered]
Hi all, thanks for releasing this great project.

I have a question regarding collision detection, what is the proper way to deal with 2 robots? How could you check for potential collisions between 2 robots in a specific joint configuration (q_robot1, q_robot2), knowing their models? It also seems that you can only consider one kinematic chain per robot, is that correct?

**Comment by balakumar-s** [comment, marked as answer]
You could treat the two robots as a single robot. We have an example here: https://curobo.org/source/getting_started/2b_isaacsim_examples.html#multi-arm-reacher

Once you set both robots to be a single robot, you can check if they are in self collision.
